### PR TITLE
[HUBDEV-2290] : Better to return proper message when missing 'requestedAccuracy' or 'address' parameter

### PR DESCRIPTION
**Pull request by dsulochana**

### CODE REVIEW

**Comment by hasitha-sandanuwan** [comment]
Can one of the admins verify this patch?

**Review by aushaniU — APPROVED** [review]

File: components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java [hunk 1]
```diff
@@ -31,43 +37,71 @@
...
-        ValidationRule[] rules = {
-                new ValidationRule(ValidationRule.VALIDATION_TYPE_MANDATORY_TEL, "address", address),
-                new ValidationRule(ValidationRule.VALIDATION_TYPE_MANDATORY_LOC_ACCURACY, "requestedAccuracy", requestAccuracy)};
-        
-        Validation.checkRequestParams(rules);
-        
-    }
+	public void validate(String[] params) throws CustomException {
+		List<String> addresses = new ArrayList<String>();
+		Double requestAccuracy = 0.0;
```
Comment (jaadds) [inline components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java]: Any reason to keep this as an Object Type. Unless specifically needed, its better to use primitive types.

File: components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java [hunk 2]
```diff
@@ -31,43 +37,71 @@
...
+		Boolean foundAddressParam = Boolean.FALSE;
```
Comment (jaadds) [inline components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java]: Same as above.

**Review by jaadds — CHANGES_REQUESTED** [review]

File: components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java (hunk 1 above)
Comment (dsulochana) [reply components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java]: Addressed review comment and fixed

File: components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java (hunk 2 above)
Comment (dsulochana) [reply components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java]: Addressed review comment and fixed

**Pull request merged**